File: benchmarks/semantic_search/fixtures/python-workflow-discovery-v1/execution/runner.py
from __future__ import annotations


def validate_workflow_parameters(
    required: set[str], supplied: dict[str, str]
) -> dict[str, str]:
    missing = required - supplied.keys()
    unknown = supplied.keys() - required
    if missing or unknown:
        raise ValueError(f"invalid workflow parameters: missing={missing}, unknown={unknown}")
    return supplied

File: benchmarks/semantic_search/fixtures/typescript-workflow-discovery-v1/execution/runner.ts
export function validateWorkflowParameters(
  required: ReadonlySet<string>,
  supplied: Record<string, string>,
): Record<string, string> {
  const missing = [...required].filter((key) => !(key in supplied));
  const unknown = Object.keys(supplied).filter((key) => !required.has(key));
  if (missing.length > 0 || unknown.length > 0) {
    throw new Error(`invalid workflow parameters: missing=${missing}, unknown=${unknown}`);
  }
  return supplied;
}
